# Generate indicator variables for disease of interests: 

### RA
### T2D
### AD

In [2]:
import os
import pandas as pd
import numpy as np

In [3]:
pwd()

'/temp_project/all_codes'

In [4]:
os.chdir('/temp_project/all_codes')
from ukbb_ldbf import load_data_by_fid

from ukbb import search_des
from ukbb import related_vars
from ukbb import mm_gen_ind_list
from ukbb import cal_vec_con

In [5]:
bmi_var_chk1=related_vars(input('enter keyword(s): ').split(','),input('check disease (T2D/RA/AD): '))
#icd, medic,illness
bmi_var_chk1


Searched keyword(s): ['bmi']
save obesity_related_vars_chk.csv file at /temp_project/ukbb/data/disease/OBESITY


fid        Type  obs_ct            Description  DC
633  21001  Continuous  499497  Body mass index (BMI) NaN
809  23104  Continuous  492465  Body mass index (BMI) NaN

In [11]:
RA_var_chk1=related_vars(input('enter keyword(s): ').split(','),input('check disease (T2D/RA/AD): '))
#icd, medic,illness
RA_var_chk1

Searched keyword(s): ['icd', ' medic', ' illness']
save RA_related_vars_chk.csv file at /temp_project/ukbb/data/disease/RA


fid                    Type  obs_ct  \
876  40001    Categorical (single)   14417   
877  40002    Categorical (single)    8768   
879  40006    Categorical (single)   69060   
886  40013    Categorical (single)   10000   
896  41076                 Integer    8716   
897  41077                 Integer       0   
898  41078                 Integer  320437   
899  41079                 Integer    3798   
907  41104                 Integer   66350   
908  41105                 Integer     176   
916  41142                 Integer  392281   
917  41143                 Integer    2733   
918  41144                 Integer   20309   
919  41145                 Integer       0   
925  41202  Categorical (multiple)  392281   
926  41203  Categorical (multiple)   20309   
927  41204  Categorical (multiple)  320437   
928  41205  Categorical (multiple)    8716   
37     137                 Integer  501740   
188   2473    Categorical (single)  501670   
189   2492    Categorical (single)  501670   
378   6153  Categorical (multiple)  270916   
379   6154  Categorical (multiple)  497893   
387   6177  Categorical (multiple)  226978   
392  10004  Categorical (multiple)    3776   
393  10005  Categorical (multiple)    3776   
400  20003  Categorical (multiple)  364052   
436  20076                    Text       0   
548  20466    Categorical (single)    7803   
581  20503    Categorical (single)    9386   
..     ...                     ...     ...   
715  22167    Categorical (single)   15493   
716  22168    Categorical (single)     365   
717  22169    Categorical (single)    1383   
718  22170    Categorical (single)    1768   
719  22171    Categorical (single)       9   
720  22172    Categorical (single)      48   
721  22173    Categorical (single)     529   
722  22174    Categorical (single)    1074   
723  22175    Categorical (single)     108   
724  22176    Categorical (single)      47   
725  22177    Categorical (single)     654   
726  22178    Categorical (single)       8   
727  22179    Categorical (single)      85   
728  22180    Categorical (single)     156   
729  22181    Categorical (single)      14   
24      87                 Integer  375111   
35     135                 Integer  501740   
168   2188    Categorical (single)  501683   
370   6145  Categorical (multiple)  497912   
399  20002  Categorical (multiple)  375111   
404  20008              Continuous  375110   
405  20009              Continuous  375110   
409  20013    Categorical (single)  375111   
444  20107    Categorical (single)  487865   
445  20110    Categorical (single)  493003   
446  20111    Categorical (single)  433990   
447  20112    Categorical (single)    5044   
448  20113    Categorical (single)    5432   
449  20114    Categorical (single)    3717   
606  20528    Categorical (single)  157363   

                                           Description        DC  
876  Underlying (primary) cause of death: ICD10Uses...      19.0  
877  Contributory (secondary) causes of death: ICD1...      19.0  
879  Type of cancer: ICD10Uses data-coding 19 compr...      19.0  
886  Type of cancer: ICD9Uses data-coding 87 compri...      87.0  
896  Episodes containing "Diagnoses - secondary ICD...       NaN  
897  Episodes containing "Diagnoses - secondary ICD...       NaN  
898  Episodes containing "Diagnoses - secondary ICD...       NaN  
899  Episodes containing "Diagnoses - secondary ICD...       NaN  
907  Episodes containing "External cause - ICD10" data       NaN  
908  Episodes containing "External cause - ICD10 ad...       NaN  
916  Episodes containing "Diagnoses - main ICD10" data       NaN  
917  Episodes containing "Diagnoses - main ICD10 - ...       NaN  
918   Episodes containing "Diagnoses - main ICD9" data       NaN  
919  Episodes containing "Diagnoses - main ICD9 - a...       NaN  
925  Diagnoses - main ICD10Uses data-coding 19 comp...      19.0  
926  Diagnoses - main ICD9Uses data-coding 87 compr...      87.0  
927  Diagnoses 

In [12]:
T2D_var_chk1=related_vars(input('enter keyword(s): ').split(','),input('check disease (T2D/RA/AD): '))
#diabet,icd,medic,illness
T2D_var_chk1

Searched keyword(s): ['diabet', 'icd', 'medic', 'illness']
save t2d_related_vars_chk.csv file at /temp_project/ukbb/data/disease/T2D


fid                    Type  obs_ct  \
185   2443    Categorical (single)  501670   
223   2976                 Integer   25288   
224   2986    Categorical (single)   25312   
277   4041    Categorical (single)   10325   
337   5890    Categorical (single)    1520   
338   5901                 Integer    1835   
378   6153  Categorical (multiple)  270916   
387   6177  Categorical (multiple)  226978   
876  40001    Categorical (single)   14417   
877  40002    Categorical (single)    8768   
879  40006    Categorical (single)   69060   
886  40013    Categorical (single)   10000   
896  41076                 Integer    8716   
897  41077                 Integer       0   
898  41078                 Integer  320437   
899  41079                 Integer    3798   
907  41104                 Integer   66350   
908  41105                 Integer     176   
916  41142                 Integer  392281   
917  41143                 Integer    2733   
918  41144                 Integer   20309   
919  41145                 Integer       0   
925  41202  Categorical (multiple)  392281   
926  41203  Categorical (multiple)   20309   
927  41204  Categorical (multiple)  320437   
928  41205  Categorical (multiple)    8716   
37     137                 Integer  501740   
188   2473    Categorical (single)  501670   
189   2492    Categorical (single)  501670   
379   6154  Categorical (multiple)  497893   
..     ...                     ...     ...   
715  22167    Categorical (single)   15493   
716  22168    Categorical (single)     365   
717  22169    Categorical (single)    1383   
718  22170    Categorical (single)    1768   
719  22171    Categorical (single)       9   
720  22172    Categorical (single)      48   
721  22173    Categorical (single)     529   
722  22174    Categorical (single)    1074   
723  22175    Categorical (single)     108   
724  22176    Categorical (single)      47   
725  22177    Categorical (single)     654   
726  22178    Categorical (single)       8   
727  22179    Categorical (single)      85   
728  22180    Categorical (single)     156   
729  22181    Categorical (single)      14   
24      87                 Integer  375111   
35     135                 Integer  501740   
168   2188    Categorical (single)  501683   
370   6145  Categorical (multiple)  497912   
399  20002  Categorical (multiple)  375111   
404  20008              Continuous  375110   
405  20009              Continuous  375110   
409  20013    Categorical (single)  375111   
444  20107    Categorical (single)  487865   
445  20110    Categorical (single)  493003   
446  20111    Categorical (single)  433990   
447  20112    Categorical (single)    5044   
448  20113    Categorical (single)    5432   
449  20114    Categorical (single)    3717   
606  20528    Categorical (single)  157363   

                                           Description        DC  
185  Diabetes diagnosed by doctorUses data-coding 1...  100349.0  
223  Age diabetes diagnosedUses data-coding 100291 ...  100291.0  
224  Started insulin within one year diagnosis of d...  100349.0  
277  Gestational diabetes onlyUses data-coding 1006...  100617.0  
337  Which eye(s) affected by diabetes-related eye ...  100515.0  
338  Age when diabetes-related eye disease diagnose...  100291.0  
378  Medication for cholesterol, blood pressure, di...  100626.0  
387  Medication for cholesterol, blood pressure or ...  100625.0  
876  Underlying (primary) cause of death: ICD10Uses...      19.0  
877  Contributory (secondary) causes of death: ICD1...      19.0  
879  Type of cancer: ICD10Uses data-coding 19 compr...      19.0  
886  Type of cancer: ICD9Uses data-coding 87 compri...      87.0  
896  Episodes containing "Diagnoses - secondary ICD...       NaN  
897  Episodes containing "Diagnoses - secondary ICD...       NaN  
898  Episodes containing "Diagnoses - secondary ICD...       NaN  
899  Episodes containing "Diagnoses - secondary ICD...       NaN  
907  Episodes c

# T2D

## Related variables check lsit





| selection lv |   index  | fid   | Type                   | obs_ct | Description                                                                                                                                                  | DC     |
|--------------|-----|-------|------------------------|--------|--------------------------------------------------------------------------------------------------------------------------------------------------------------|--------|
| 1            | 185 | 2443  | Categorical (single)   | 501670 | Diabetes diagnosed by doctorUses data-coding 100349 comprises 4 Integer-valued members in a simple list.                                                     | 100349 |
| 0.5          | 223 | 2976  | Integer                | 25288  | Age diabetes diagnosedUses data-coding 100291 comprises 2 Integer-valued members in a simple list.                                                           | 100291 |
| 0.5          | 224 | 2986  | Categorical (single)   | 25312  | Started insulin within one year diagnosis of diabetesUses data-coding 100349 comprises 4 Integer-valued members in a simple list.                            | 100349 |
| 0.5          | 277 | 4041  | Categorical (single)   | 10325  | Gestational diabetes onlyUses data-coding 100617 comprises 5 Integer-valued members in a simple list.                                                        | 100617 |
| 0.5          | 337 | 5890  | Categorical (single)   | 1520   | Which eye(s) affected by diabetes-related eye diseaseUses data-coding 100515 comprises 3 Integer-valued members in a simple list.                            | 100515 |
| 0.5          | 338 | 5901  | Integer                | 1835   | Age when diabetes-related eye disease diagnosedUses data-coding 100291 comprises 2 Integer-valued members in a simple list.                                  | 100291 |
| 1            | 378 | 6153  | Categorical (multiple) | 270916 | Medication for cholesterol, blood pressure, diabetes, or take exogenous hormonesUses data-coding 100626 comprises 8 Integer-valued members in a simple list. | 100626 |
| 1            | 387 | 6177  | Categorical (multiple) | 226978 | Medication for cholesterol, blood pressure or diabetesUses data-coding 100625 comprises 6 Integer-valued members in a simple list.                           | 100625 |
| 0.5          | 876 | 40001 | Categorical (single)   | 14417  | Underlying (primary) cause of death: ICD10Uses data-coding 19 comprises 19154 String-valued members in a hierarchical tree.                                  | 19     |
| 0.5          | 877 | 40002 | Categorical (single)   | 8768   | Contributory (secondary) causes of death: ICD10Uses data-coding 19 comprises 19154 String-valued members in a hierarchical tree.                             | 19     |
| 0.5          | 879 | 40006 | Categorical (single)   | 69060  | Type of cancer: ICD10Uses data-coding 19 comprises 19154 String-valued members in a hierarchical tree.                                                       | 19     |
| 0.5          | 886 | 40013 | Categorical (single)   | 10000  | Type of cancer: ICD9Uses data-coding 87 comprises 13710 String-valued members in a hierarchical tree.                                                        | 87     |
| 1            | 925 | 41202 | Categorical (multiple) | 392281 | Diagnoses - main ICD10Uses data-coding 19 comprises 19154 String-valued members in a hierarchical tree.                                                      | 19     |
| 1            | 926 | 41203 | Categorical (multiple) | 20309  | Diagnoses - main ICD9Uses data-coding 87 comprises 13710 String-valued members in a hierarchical tree.                                                       | 87     |
| 1            | 927 | 41204 | Categorical (multiple) | 320437 | Diagnoses - secondary ICD10Uses data-coding 19 comprises 19154 String-valued members in a hierarchical tree.                                                 | 19     |
| 1            | 928 | 41205 | Categorical (multiple) | 8716   | Diagnoses - secondary ICD9Uses data-coding 87 comprises 13710 String-valued members in a hierarchical tree.                                                  | 87     |
| 2            | 400 | 20003 | Categorical (multiple) | 364052 | Treatment/medication codeUses data-coding 4 comprises 6745 Integer-valued members in a simple list.                                                          | 4      |
| 0.5          | 436 | 20076 | Text                   | 0      | Treatment/medication READ-code                                                                                                                               |        |
| 1.5          | 399 | 20002 | Categorical (multiple) | 375111 | Non-cancer illness code, self-reportedUses data-coding 6 comprises 474 Integer-valued members in a hierarchical tree.                                        | 6      |
| 0.1          | 444 | 20107 | Categorical (single)   | 487865 | Illnesses of fatherUses data-coding 1010 comprises 18 Integer-valued members in a simple list.                                                               | 1010   |
| 0.1          | 445 | 20110 | Categorical (single)   | 493003 | Illnesses of motherUses data-coding 1010 comprises 18 Integer-valued members in a simple list.                                                               | 1010   |
| 0.1          | 446 | 20111 | Categorical (single)   | 433990 | Illnesses of siblingsUses data-coding 1010 comprises 18 Integer-valued members in a simple list.                                                             | 1010   |
| 0.1          | 447 | 20112 | Categorical (single)   | 5044   | Illnesses of adopted fatherUses data-coding 1010 comprises 18 Integer-valued members in a simple list.                                                       | 1010   |
| 0.1          | 448 | 20113 | Categorical (single)   | 5432   | Illnesses of adopted motherUses data-coding 1010 comprises 18 Integer-valued members in a simple list.                                                       | 1010   |
| 0.1          | 449 | 20114 | Categorical (single)   | 3717   | Illnesses of adopted siblingsUses data-coding 1010 comprises 18 Integer-valued members in a simple list.                                                     | 1010   |

|dis|ICD-10 | ICD-9|
|---|
|T1D|E10.x|25001, 25011, 25021, 25091|
|T2D|E11.x|25000, 25010, 25020, 25090|
|T2Dex|E111,E116,E131|25010, 25012|

|Keyword(s)|fid|Description|code|
|---|
|diabete, icd, medic, illness|2443|Diabetes diagnosed by doctor|1=Yes|
||6153|Medication for cholesterol, blood pressure, diabetes, or take exogenous hormones|3=Insulin|
||6177|Medication for cholesterol, blood pressure or diabetes|3=Insulin|
||41202|ICD-10 main|T1D:E10x ----   T2D:E11x|
||41204|ICD-10 2nd|T1D:E10x ----  T2D:E11x|
||41203|ICD-9 main|T1D: 25001, 25011, 25021, 25091  ----  T2D: 25001, 25011, 25021, 25091|
||41205|ICD-9 2nd|T1D: 25001, 25011, 25021, 25091  ----  T2D: 25000, 25010, 25020, 25090|
||20002|self-reported|T1D:1222   T2D:1223|
|||||
|||||